### PR TITLE
backport: merge bitcoin#28577, #28588, #28634, #28849, #28805, #28951, #29058, #29239, partial bitcoin#28331, #28452 (BIP324 backports: part 2)

### PR DESCRIPTION
**Pull request by kwvg**
## Additional Information

* Dependent on https://github.com/dashpay/dash/pull/6280

* Dependent on https://github.com/dashpay/dash/pull/6276

* Dependency for https://github.com/dashpay/dash/pull/6329

* [bitcoin#28452](https://github.com/bitcoin/bitcoin/pull/28452) was backported as the behavior it introduces is required for backports like [bitcoin#28331](https://github.com/bitcoin/bitcoin/pull/28331). As it pre-dates earlier BIP324 backports, the `ClearShrink` usage from those backports have also been incorporated into this backport.

* When backporting [bitcoin#28331](https://github.com/bitcoin/bitcoin/pull/28331), in TestFramework's `add_nodes()`, `extra_args[i]` is not converted to a `list` like it is done upstream and avoiding duplication of `-v2transport=1` is instead done by an additional check. This is done to prevent test failures in `feature_mnehf.py`.

* The local variable `args` is removed in [bitcoin#28805](https://github.com/bitcoin/bitcoin/pull/28805) as it does nothing (the argument appending logic is removed as part of the backport) and upstream removes it anyways. 

Special thanks to UdjinM6 for significant contributions to this and dependent PRs! 🎉 

## Breaking Changes

None expected.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas **(note: N/A)**
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Review by UdjinM6 — COMMENTED** [review]
LGTM in general and seems to be working on testnet, few notes:
- 28331: missing changes in `src/rpc/util.cpp`, should be partial
- needs release notes I guess (for the addition of the protocol itself and for changes in cli output, cmd-line options and RPCs)
- should we do a proto bump?

**Comment by kwvg** [comment]
@UdjinM6 

> * 28331: missing changes in src/rpc/util.cpp, should be partial

Resolved in latest push, missing change now noted in commit comment.

> * needs release notes I guess (for the addition of the protocol itself and for changes in cli output, cmd-line options and RPCs)

We can take reference from upstream v26 ([source](https://github.com/bitcoin/bitcoin/blame/48cf3da636089873ba7280e0d5b22eb81811d194/doc/release-notes/release-notes-26.0.md#L46-L50)) release notes but would prefer to do that in a separate PR.

> * should we do a proto bump?

IMO that isn't necessary, BIP324/P2Pv2 is a change in the transport layer, the protocol layer is more-or-less unaffected and P2Pv2 can gracefully fallback to P2Pv1.

File: test/functional/p2p_v2_transport.py [hunk 1]
```diff
@@ -148,6 +148,19 @@ def run_test(self):
...
 
+        # Check detection of missing garbage terminator (hits after fixed amount of data if terminator never matches garbage)
+        MAX_KEY_GARB_AND_GARBTERM_LEN = 64 + 4095 + 16
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            num_peers = len(self.nodes[0].getpeerinfo())
+            s.connect(("127.0.0.1", p2p_port(0)))
+            self.wait_until(lambda: len(self.nodes[0].getpeerinfo()) == num_peers + 1)
+            s.sendall(b'\x00' * (MAX_KEY_GARB_AND_GARBTERM_LEN - 1))
+            self.wait_until(lambda: self.nodes[0].getpeerinfo()[-1]["bytesrecv"] == MAX_KEY_GARB_AND_GARBTERM_LEN - 1)
+            with self.nodes[0].assert_debug_log("V2 transport error: missing garbage terminator"):
```
[the remark this answers is not in the document]
Comment (kwvg) [reply test/functional/p2p_v2_transport.py:159]: This is a known bug that has been resolved in bitcoin#28645, which is a part of the next set of BIP324 backports

File: src/bitcoin-cli.cpp [hunk 2]
```diff
@@ -517,10 +518,11 @@ class NetinfoRequestHandler : public BaseRequestHandler
                 const std::string addr{peer["addr"].get_str()};
                 const std::string age{conn_time == 0 ? "" : ToString((m_time_now - conn_time) / 60)};
                 const std::string sub_version{peer["subver"].get_str()};
+                const std::string transport{peer["transport_protocol_type"].get_str()};
```
[the remark this answers is not in the document]
Comment (kwvg) [reply src/bitcoin-cli.cpp:521]: Good suggestion! Will add in next set of upcoming backports (its emphasis is specifically fleshing out tests, UI and dealing with intermittent failures and edge cases, would fit better there)

**Review by PastaPastaPasta — APPROVED** [review]
utACK 5dd60c48757ba552796f613e261b257a5df2453d
